packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -76,12 +100,41 @@ export const useStoreSwitcher = props => {
         );
     }, [storeConfigData, availableStoresData]);
 
+    // Get suffix based on page type
+    const getSuffix = useCallback(
+        storeCode => {
+            let suffix = '';
+            if (pageType === 'CATEGORY') {
+                suffix = availableStores.get(storeCode).categorySuffix || '';
+            }
+            if (pageType === 'PRODUCT') {
+                suffix = availableStores.get(storeCode).productSuffix || '';
+            }
+
+            return suffix;
+        },
+        [availableStores, pageType]
+    );
+
+    // Handle updating the URL with configured product or catalog url suffix
+    const setNewUrl = useCallback(
```

Comment:
I'd name this something else - it's not actually setting anything, just making a string.

In fact you could probably do away with this function and just put lines 122-125 in the `else` of the switch callback.